examples/cli/readme.md

```diff
@@ -1 +1,619 @@
+
+Sometimes you will be interested to analyze or benchmark those sub-modules, which is where the `--max-depth` argument comes in.
+
+For example, if you run this command:
+
+```
+benchit benchmark scripts/max_depth.py
+```
+
+You will get a result that looks very similar to the [Hello World tutorial](#hello-world) tutorial. However, if you peek into `max_depth.py`, you can see that there are two `torch.nn.Linear` modules that make up the top-level model.
+
+You can analyze and benchmark those `torch.nn.Linear` modules with this command:
+
+```
+benchit benchmark scripts/max_depth.py --max-depth 1
+```
+
+You get a result like:
+
+```
+Models discovered during profiling:
+
+max_depth.py:
+        pytorch_model (executed 1x)
+                Model Type:     Pytorch (torch.nn.Module)
+                Class:          TwoLayerModel (<class 'max_depth.TwoLayerModel'>)
+                Location:       /home/jfowers/mlagility/examples/cli/scripts/max_depth.py, line 41
+                Parameters:     85 (<0.1 MB)
+                Hash:           80b93950
+                Status:         Model successfully benchmarked on Intel(R) Xeon(R) CPU @ 2.20GHz
+                                Mean Latency:   0.000   milliseconds (ms)
+                                Throughput:     533884.4        inferences per second (IPS)
+
+max_depth.py:
+                        fc (executed 2x)
+                                Model Type:     Pytorch (torch.nn.Module)
+                                Class:          Linear (<class 'torch.nn.modules.linear.Linear'>)
+                                Parameters:     55 (<0.1 MB)
+                                Hash:           6d5eb4ee
+                                Status:         Model successfully benchmarked on Intel(R) Xeon(R) CPU @ 2.20GHz
+                                                Mean Latency:   0.000   milliseconds (ms)
+                                                Throughput:     809701.4        inferences per second (IPS)
+
+                        fc2 (executed 2x)
+                                Model Type:     Pytorch (torch.nn.Module)
+                                Class:          Linear (<class 'torch.nn.modules.linear.Linear'>)
+                                Parameters:     30 (<0.1 MB)
+                                Hash:           d4b2ffa7
+                                Status:         Model successfully benchmarked on Intel(R) Xeon(R) CPU @ 2.20GHz
+                                                Mean Latency:   0.000   milliseconds (ms)
+                                                Throughput:     677945.2        inferences per second (IPS)
+```
+
+You can see that the two instances of `torch.nn.Linear`, `fc` and `fc2`, are benchmarked in addition to the top-level model, `pytorch_model`.
+
+> See the [Max Depth documentation](https://github.com/groq/mlagility/blob/main/tools_user_guide.md#maximum-analysis-depth) for more details.
```

Comment:
This should go to the advanced tutorial (if we decide that having one is a good idea)

Reply:
Done! It's in `discovery.md` now.